frontend/src/stimulus/controllers/dynamic/work-packages/progress/preview-progress.controller.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * -- copyright
+ * OpenProject is an open source project management software.
+ * Copyright (C) 2024 the OpenProject GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License version 3.
+ *
+ * OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+ * Copyright (C) 2006-2013 Jean-Philippe Lang
+ * Copyright (C) 2010-2013 the ChiliProject Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * See COPYRIGHT and LICENSE files for more details.
+ * ++
+ */
+
+import { Controller } from '@hotwired/stimulus';
+import { debounce } from 'lodash';
+import morphdom from 'morphdom';
+
+interface TurboBeforeFrameRenderEventDetail {
+  render:(currentElement:HTMLElement, newElement:HTMLElement) => void;
+}
+
+export default class PreviewProgressController extends Controller {
+  static targets = [
+    'form', 'progressInput',
+  ];
+
+  declare readonly progressInputTargets:HTMLInputElement[];
+  declare readonly formTarget:HTMLFormElement;
+
+  private debouncedPreview:(event:Event) => void;
+  private frameMorphRenderer:(event:CustomEvent<TurboBeforeFrameRenderEventDetail>) => void;
+
+  connect() {
+    this.debouncedPreview = debounce((event:Event) => { void this.preview(event); }, 500);
```

Comment:
I would reduce the debounce time as it feels a bit sluggish. On the other hand, when reducing the time, I oftentimes ran into the situation where my last input would be eaten up. Knowing of this problem, even when not changing the time, I could time my input so that they would be eaten up. I don't know if it is just a forced scenario or something that is likely to happen to users that just type in numbers at the wrong time. Since the inserted values normally shouldn't be that long I would think that it is less likely to happen so I would be willing to accept this potential problem and also the need for the sluggy behaviour.

Reply:
In datepickers, we have a similar pattern and also use a debounce of 500: https://github.com/opf/openproject/blob/dev/frontend/src/app/shared/components/datepicker/wp-multi-date-form/wp-multi-date-form.component.ts#L802

So I'd leave it as it is and fine-tune later